### PR TITLE
[ARTS-689][Subfeature] Init use site service client

### PR DESCRIPTION
**Pull request by anatbal**
## Description
- Init uses the new site-service-client

### Changes
- [ARTS-689] - Init uses the new site-service-client
### Checklist:

- [x] Branch name follows convention (feature/arts-#-short-name OR fix/arts-#-short-name)
- [x] I have included a short-meaningful title, description and changes for this PR

[ARTS-689]: https://ndph-arts.atlassian.net/browse/ARTS-689

### CODE REVIEW

**Review by shajidissa — DISMISSED** [review]

File: init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/loader/Loader.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ public void run(String... args) throws Exception {
             var sites = trialConfig.getSites();
 
             initProgressReporter.submitProgress(LoaderProgress.CREATE_SITES.message());
-            List<String> siteIds = siteServiceInvoker.execute(sites);
+            List<String> siteIds = siteServiceInvoker.createManySites(sites);
```
Comment (darrengsmithson) [inline init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/loader/Loader.java:86]: why createManySites? .createSites would be the usual approach? Do we expect there to be a createNotManySites ?

**Review by darrengsmithson — CHANGES_REQUESTED** [review]

File: init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/loader/Loader.java (hunk 1 above)
Comment (anatbal) [reply init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/loader/Loader.java]: I agree, changed it to .createSites. In addition, to keep the convention I change createManyRoles to createRoles as well in the other invoker, so these files were added to this PR but it is only about name changing ang nothing more

File: init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/model/Trial.java [hunk 2]
```diff
@@ -19,7 +20,7 @@
     @NotEmpty
     private List<@Valid Practitioner> persons;
     @NotEmpty
-    private List<@Valid Site> sites;
+    private List<@Valid SiteDTO> sites;
```
Comment (eladiw) [inline init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/model/Trial.java:23]: you need to re run the build tool schema builder and publish updated schemas in the second repo

**Review by eladiw — APPROVED** [review]

**Review by darrengsmithson — APPROVED** [review]

File: init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/model/Trial.java (hunk 2 above)
Comment (anatbal) [reply init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/model/Trial.java:23]: Will do. Need to discuss how to make validation for the clients models

**Pull request merged**